gossip/mempool.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+// Tx is a transaction that can be gossiped across the network.
+type Tx interface {
+	// Hash returns the unique hash of this transaction
+	Hash() []byte
+	// Marshal returns the byte representation of this transaction
+	Marshal() ([]byte, error)
+	// Unmarshal deserializes the provided bytes in-place
+	Unmarshal(b []byte) error
+}
+
+// Mempool is holds pending transactions
+type Mempool[T Tx] interface {
+	// AddTx adds a transaction to the mempool
+	AddTx(tx T) error
+	// PendingTxs returns pending transactions in the mempool
+	GetPendingTxs() []T
```

Comment:
optional: we can make this fn take a bloom filter and filter out what is already contained in the filter.

Reply:
I think we could do something like that - I'll add a `filter func(tx T) bool` instead so it's a bit more flexible, and we can still pass in the bloom filter from the caller.